src/main/java/io/zonky/test/db/postgres/embedded/LiquibasePreparer.java

```diff
@@ -50,8 +61,14 @@ public void prepare(DataSource ds) throws SQLException {
         try {
             connection = ds.getConnection();
             Database database = getInstance().findCorrectDatabaseImplementation(new JdbcConnection(connection));
-            Liquibase liquibase = new Liquibase(location, new ClassLoaderResourceAccessor(), database);
+
+            Liquibase liquibase = new Liquibase(location, accessor, database);
             liquibase.update(contexts);
+            
+            // Try and commit the changes.
+            connection.commit();
```

Comment:
Is this change necessary? I would prefer to keep it as it was. I guess the calling of `liquibase.update(contexts)` should commit all the changes, right? And the closing logic is already implemented in the finally block.

Reply:
Unconditionally rolling back in the `finally` seems incorrect. Closing without committing seems risky, so I added commit + close + set to null

Reply:
You're right that the previous rolling back logic was incorrect. I've replaced it with a try-with-resources block.